src/emit/instructions.rs

```diff
@@ -292,14 +293,18 @@ pub(super) fn process_instruction<'a, T: TargetRuntime<'a> + ?Sized>(
 
             bin.builder.position_at_end(error);
             bin.log_runtime_error(target, "pop from empty array".to_string(), Some(*loc), ns);
-            target.assert_failure(
-                bin,
-                bin.context
-                    .i8_type()
-                    .ptr_type(AddressSpace::default())
-                    .const_null(),
-                bin.context.i32_type().const_zero(),
-            );
+            let (revert_out, revert_out_len) = if ns.target == Target::Solana {
+                (
+                    bin.context
+                        .i8_type()
+                        .ptr_type(AddressSpace::default())
+                        .const_null(),
+                    bin.context.i32_type().const_zero(),
+                )
+            } else {
+                bin.panic_data_const(ns, PanicCode::EmptyArrayPop)
+            };
+            target.assert_failure(bin, revert_out, revert_out_len);
```

Comment:
This sequence of code is repeated a few times, might be good to factor out into a function (maybe something on target?)

Reply:
Good point, I'll look into it

Reply:
I moved it into `panic_data_const`